drivers/net/ethernet/renesas/rswitch.c

```diff
@@ -2741,6 +2750,7 @@ static void rswitch_fib_event_remove(struct rswitch_fib_event_work *fib_work)
 	if (!dev)
 		return;
 
+	mutex_lock(&dev->priv->ipv4_forward_lock);
```

Comment:
It seems like this lock can be redundant because forward flag status has to be kept in mind only when you setup hw.